### DISCUSSION TITLE
Does NirCmd actually work?

### DISCUSSION THREAD

**Discussion by eggbean** [ScoopInstaller/Scoop#5768] [Help and Questions] [answered]
I have the `nirsoft` bucket added and I have installed `[nircmd](https://nircmd.nirsoft.net/shortcut.html)`, which I think is single binary that has a lot of commands, so I expected scoop to make a bunch of shims for it, but it doesn't appear to have done so.

I want to use the `shortcut` command that it has but typing that doesn't do anything and nor does typing `nircmd`.

How do I use it? Is it broken?

**Comment by HUMORCE** [comment]
Its name `nircmd` is ignored in commands reference of online document and attached help file.

you should use the commands by execute `nircmd <command> args..`. for example, If you want to use `beep` then execute `nircmd beep [Frequency] [Duration]`.

---

> I expected scoop to make a bunch of shims for it

You can create shims by yourself, `scoop help shim`

**Reply by eggbean** [reply]
Thanks, but there doesn't seem to be a shim created for `nircmd.exe` itself. I think there might be an error.

**Reply by HUMORCE** [reply, marked as answer]
https://github.com/ScoopInstaller/Nirsoft/blob/master/bucket/nircmd.json

`nircmd` listed in the manifest, try `scoop reset nircmd`, you can see the creation.

**Reply by eggbean** [reply]
Okay thanks, but what actually worked was re-attempting to install it, which I had tried before (I uninstalled and reinstalled it), but this time it worked somehow.

```
F:\Users\jason\winfiles\bin>scoop install nircmd
WARN  Purging previous failed installation of nircmd.
ERROR You need admin rights to uninstall global apps.
WARN  Purging previous failed installation of nircmd.
ERROR 'nircmd' isn't installed correctly.
'nircmd' was uninstalled.
Installing 'nircmd' (2.86) [64bit] from main bucket
nircmd-x64.zip (152.1 KB) [==============================================================================================================================================================================] 100%
Checking hash of nircmd-x64.zip ... ok.
Extracting nircmd-x64.zip ... done.
Linking C:\ProgramData\scoop\apps\nircmd\current => C:\ProgramData\scoop\apps\nircmd\2.86
'nircmd' (2.86) was installed successfully!
```

(EDIT: I have uninstalled it again and reinstalled it using `gsudo` now, as I forgot I was using a shared directory.)

**Reply by eggbean** [reply]
While it is working now (I can use it), it doesn't come up in the shim list.

    scoop shim list | grep -i nircmd

This returns nothing, and I looked through the whole ungrepped list too.

I am using shared directory though.

    F:\Users\jason\winfiles\bin>echo %SCOOP%
    C:\ProgramData\scoop

I've been using scoop for years and haven't had any problem like this before, but only recently started using a shared directory between different accounts.

Other than this strange problem, the only other thing is that everything gets listed twice in the install and shim lists. This causes problems with `WingetUI` which I find very useful for updating all scoop, winget, Windows Store apps and even programs that were installed manually. Since I started using this shared scoop directory I haven't used `WingetUI` to update everything until I figure out what to do abou this issue.

**Reply by HUMORCE** [reply]
> This returns nothing, and I looked through the whole ungrepped list too.

emmmm, because `main/nircmd` was installed instead of `nirsoft/nircmd`. I did not noticed that before.

`main/nircmd`: adds the app root dir to the PATH.
`nirsoft/nircmd`: creates shim/shortcut for nircmd.exe.

**Reply by eggbean** [reply]
I don't understand why there are no shims when the package is from the `main` bucket? In my previous installation (for single user) I don't remember having these issues?

I am now finding that there are shims missing for `touch` and `tail` (but not `head`), while I have there different packages installed which should have these commands.

https://github.com/ScoopInstaller/Scoop/discussions/5770

**Reply by HUMORCE** [reply]
See https://github.com/ScoopInstaller/Main/pull/4212
